filter/rastertolabel.c

```diff
@@ -492,144 +530,75 @@ EndPage(ppd_file_t          *ppd,	/* I - PPD file */
         puts("^XA");
 
        /*
-        * Rotate 180 degrees so that the top of the label/page is at the
-	* leading edge...
+	* Emit label header...
 	*/
 
-	puts("^POI");
+	ZPLEmitLabelHeader(ppd, header);
 
        /*
-        * Set print width...
+        * Display the label image...
 	*/
 
-        printf("^PW%u\n", header->cupsWidth);
+	puts("^FO0,0^XGR:CUPS.GRF,1,1^FS");
 
        /*
-        * Set print rate...
+        * End the label and eject...
 	*/
 
-	if ((choice = ppdFindMarkedChoice(ppd, "zePrintRate")) != NULL &&
-	    strcmp(choice->choice, "Default"))
-	{
-	  val = atoi(choice->choice);
-	  printf("^PR%d,%d,%d\n", val, val, val);
-	}
+	puts("^XZ");
 
        /*
-        * Put label home in default position (0,0)...
+        * Delete the label image...
         */
 
-	printf("^LH0,0\n");
-
-       /*
-        * Set media tracking...
-	*/
-
-	if (ppdIsMarked(ppd, "zeMediaTracking", "Continuous"))
-	{
-         /*
-	  * Add label length command for continuous...
-	  */
-
-	  printf("^LL%d\n", header->cupsHeight);
-	  printf("^MNN\n");
-	}
-	else if (ppdIsMarked(ppd, "zeMediaTracking", "Web"))
-          printf("^MNY\n");
-	else if (ppdIsMarked(ppd, "zeMediaTracking", "Mark"))
-	  printf("^MNM\n");
-
-       /*
-        * Set label top
-	*/
-
-	if (header->cupsRowStep != 200)
-	  printf("^LT%d\n", header->cupsRowStep);
-
-       /*
-        * Set media type...
-	*/
-
-	if (!strcmp(header->MediaType, "Thermal"))
-	  printf("^MTT\n");
-	else if (!strcmp(header->MediaType, "Direct"))
-	  printf("^MTD\n");
-
-       /*
-        * Set print mode...
-	*/
-
-	if ((choice = ppdFindMarkedChoice(ppd, "zePrintMode")) != NULL &&
-	    strcmp(choice->choice, "Saved"))
-	{
-	  printf("^MM");
-
-	  if (!strcmp(choice->choice, "Tear"))
-	    printf("T,Y\n");
-	  else if (!strcmp(choice->choice, "Peel"))
-	    printf("P,Y\n");
-	  else if (!strcmp(choice->choice, "Rewind"))
-	    printf("R,Y\n");
-	  else if (!strcmp(choice->choice, "Applicator"))
-	    printf("A,Y\n");
-	  else
-	    printf("C,Y\n");
-	}
-
-       /*
-        * Set tear-off adjust position...
-	*/
-
-	if (header->AdvanceDistance != 1000)
-	{
-	  if ((int)header->AdvanceDistance < 0)
-	    printf("~TA%04d\n", (int)header->AdvanceDistance);
-	  else
-	    printf("~TA%03d\n", (int)header->AdvanceDistance);
-	}
-
-       /*
-        * Allow for reprinting after an error...
-	*/
-
-	if (ppdIsMarked(ppd, "zeErrorReprint", "Always"))
-	  printf("^JZY\n");
-	else if (ppdIsMarked(ppd, "zeErrorReprint", "Never"))
-	  printf("^JZN\n");
-
-       /*
-        * Print multiple copies
-	*/
-
-	if (header->NumCopies > 1)
-	  printf("^PQ%d, 0, 0, N\n", header->NumCopies);
+	puts("^XA");
+        puts("^IDR:CUPS.GRF^FS");
+	puts("^XZ");
 
        /*
-        * Display the label image...
+        * Cut the label as needed...
```

Comment:
According developing README every 8 spaces have to be 1 tab. Please apply the same on other places.